bench_test.go

```diff
@@ -0,0 +1,109 @@
+package frostdb
+
+import (
+	"context"
+	"testing"
+
+	"github.com/apache/arrow/go/v8/arrow"
+	"github.com/apache/arrow/go/v8/arrow/memory"
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+
+	"github.com/polarsignals/frostdb/query"
+	"github.com/polarsignals/frostdb/query/logicalplan"
+)
+
+const (
+	dbName      = "parca"
+	tableName   = "stacktraces"
+	storagePath = "testdata/data"
+)
+
+// This file runs end-to-end benchmarks on a single FrostDB instance using
+// Polar Signals-specific data and schema. These benchmarks are skipped by
+// default since they require a manually-generated data directory to run. To run
+// these benchmarks, unskip the benchmarks and create a data directory in
+// testdata/data.
+
+func newDBForBenchmarks(ctx context.Context, b *testing.B) (*DB, error) {
+	b.Helper()
+
+	b.Logf("initializing %s", b.Name())
+
+	col, err := New(
+		log.NewNopLogger(),
+		prometheus.NewRegistry(),
+		WithWAL(),
+		WithStoragePath(storagePath),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := col.ReplayWALs(ctx); err != nil {
+		return nil, err
+	}
+
+	colDB, err := col.DB(ctx, dbName)
+	if err != nil {
+		return nil, err
+	}
+
+	table, err := colDB.GetTable(tableName)
+	if err != nil {
+		return nil, err
+	}
+	table.Sync()
+
+	b.Logf("db initialized and WAL replayed, starting benchmark %s", b.Name())
+	return colDB, nil
+}
+
+// Remove unused warning.
+var _ = newDBForBenchmarks
+
+func BenchmarkTypes(b *testing.B) {
+	b.Skip()
+
+	ctx := context.Background()
+	db, err := newDBForBenchmarks(ctx, b)
+	require.NoError(b, err)
+
+	engine := query.NewEngine(memory.NewGoAllocator(), db.TableProvider())
+
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		if err := engine.ScanTable(tableName).
+			Distinct(
+				logicalplan.Col("name"),
+				logicalplan.Col("sample_type"),
+				logicalplan.Col("sample_unit"),
+				logicalplan.Col("period_type"),
+				logicalplan.Col("period_unit"),
+				logicalplan.Col("duration").Gt(logicalplan.Literal(0)),
+			).
+			Execute(ctx, func(r arrow.Record) error {
+				return nil
+			}); err != nil {
+			b.Fatalf("query returned error: %v", err)
+		}
+	}
+}
+
+// BenchmarkMerge executes a merge of profiles over a 15-minute time window.
+func BenchmarkMerge(b *testing.B) {
+	b.Skip()
```

Comment:
`nit` prefer to put the comment below into the skip reason unless you plan to wait to merge until the below is satisfied